### DISCUSSION TITLE
IQueryable / LINQ support

### DISCUSSION THREAD

**Discussion by OneCyrus** [Giorgi/DuckDB.NET#109] [General]
Did you look already into creating a IQueryable/LINQ provider? There are many use cases in .NET where such a provider would be really nice. 

We are currently looking into enabling easy access to parquet files over an GraphQL API. IQueryable would allow us to filter and project client requests to duckdb directly.

**Comment by Giorgi** [comment]
Implementing an EF Core provider for a database is a huge task. I don't have any (short-term) plans to implement it. You could probably modify [EF Core SQLite](https://github.com/dotnet/efcore/tree/main/src/EFCore.Sqlite.Core) provider and modify it to work with DuckDB instead but I haven't looked into it.

If you need the provider to use in a commercial product, you can contribute it back to the project or you could fund the development.

**Reply by garethpotter** [reply]
We've built [Ducklinq](https://github.com/Brill-Power/Ducklinq/), a rudimentary LINQ-to-DuckDb provider (not on top of EF Core). It's a work in progress, but perhaps it will be useful to someone.

**Comment by VikasAtEnEx** [comment]
We've added read-only DuckDB support for Entity Framework Core, enabling integration while preserving the standard EF Core developer experience.
This new adapter offers a straightforward UseDuckDb() API (also available as UseDuckDbOnParquet()), enabling support for both .duckdb files and Parquet datasets. It internally manages SQL translation and works out-of-the-box with EF Core features such as LINQ queries, navigation properties, and dependency injection-based configuration—exclusively for read operations at this stage.
Performance-wise, this integration achieves up to 8 to 10 times faster query speeds compared to SQLite for analytical workloads, with a simplified setup that requires no manual connection handling or interceptors.
⚡ With EF Core compatibility in place, you can easily layer OData or GraphQL on top to expose flexible, queryable APIs over DuckDB or Parquet files—ideal for data exploration and lightweight analytics scenarios.
⚠️ Note: Write operations are not supported in this initial release.

https://github.com/EnergyExemplar/EnergyExemplar.EFCore.Duckdb.Extensions
